### PR TITLE
Change hasRemoteParent to be Boolean where null means root Span.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by adriancole** [comment]
Mention why null parent doesnt imply the same? If it does then maybe add to docs that null here is correlated

File: api/src/main/java/io/opencensus/trace/Sampler.java [hunk 1]
```diff
@@ -21,9 +21,10 @@
   /**
    * Called during {@link Span} creation to make a sampling decision.
    *
-   * @param parentContext The parent {@code Span} {@link SpanContext}. May be {@code null} if this
-   *     is a root span.
-   * @param remoteParent true if the parentContext is remote.
+   * @param parentContext The parent {@code Span} {@link SpanContext}. {@code null} if this is a
```
Comment (g-easy) [inline api/src/main/java/io/opencensus/trace/Sampler.java]: Optional: the parent span's spancontext

File: impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
   private SpanImpl startSpanInternal(
       @Nullable SpanContext parent,
-      boolean hasRemoteParent,
+      Boolean hasRemoteParent,
```
Comment (g-easy) [inline impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java]: @Nullable?

File: impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java [hunk 3]
```diff
@@ -123,12 +123,13 @@ private SpanBuilderImpl(
...
-    boolean hasRemoteParent = parentContext != null;
+    Boolean hasRemoteParent = Boolean.TRUE;
     TimestampConverter timestampConverter = null;
-    if (!hasRemoteParent) {
+    if (remoteParentSpanContext == null) {
       // This is not a child of a remote Span. Get the parent SpanContext from the parent Span if
       // any.
       Span parent = this.parent;
+      hasRemoteParent = Boolean.FALSE;
       if (parent != null) {
```
Comment (g-easy) [inline impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java]: else hasRemoteParent = null?

File: impl_core/src/main/java/io/opencensus/trace/SpanImpl.java [hunk 4]
```diff
@@ -44,7 +44,7 @@
   // The parent SpanId of this span. Null if this is a root span.
   private final SpanId parentSpanId;
   // True if the parent is on a different process.
-  private final boolean hasRemoteParent;
+  private final Boolean hasRemoteParent;
```
Comment (g-easy) [inline impl_core/src/main/java/io/opencensus/trace/SpanImpl.java]: Does this need to be marked Nullable?

Comment (bogdandrutu) [inline impl_core/src/main/java/io/opencensus/trace/SpanImpl.java]: Use the annotation only for public API.

File: impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java (hunk 3 above)
Comment (bogdandrutu) [inline impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java]: Done.

File: impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java (hunk 2 above)
Comment (bogdandrutu) [inline impl_core/src/main/java/io/opencensus/trace/SpanBuilderImpl.java]: Done.

File: api/src/main/java/io/opencensus/trace/Sampler.java (hunk 1 above)
Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/Sampler.java]: Done.

File: api/src/main/java/io/opencensus/trace/export/SpanData.java [hunk 5]
```diff
@@ -44,7 +44,8 @@
    * @param context the {@code SpanContext} of the {@code Span}.
    * @param parentSpanId the parent {@code SpanId} of the {@code Span}. {@code null} if the {@code
    *     Span} is a root.
-   * @param hasRemoteParent {@code true} if the parent is on a different process.
+   * @param hasRemoteParent {@code true} if the parentContext is remote. {@code null} if this is a
```
Comment (g-easy) [inline api/src/main/java/io/opencensus/trace/export/SpanData.java]: I don't see anything called parentContext in this file.

**Pull request merged**